api_server/src/main/java/io/zetch/app/repo/LocationRepository.java

```diff
@@ -5,12 +5,23 @@
 import java.util.List;
 import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.lang.Nullable;
 
 /** Backend for the Location data. */
 public interface LocationRepository extends JpaRepository<LocationEntity, Long> {
   boolean existsByName(String name);
 
   Optional<LocationEntity> findByName(String name);
 
-  List<LocationEntity> findByNameAndType(String name, Type type);
+  @Query(
+      """
+      select l from LocationEntity l
+      where upper(l.name) like upper(concat('%', :name, '%')) or
```

Comment:
What happens if a user doesn't provide name or description? It'd be true for every location entities??

Reply:
I tested it, and if name or description are null, then it does not evaluate to true